presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.common.Page;
+import com.facebook.presto.common.PageBuilder;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.google.common.collect.ImmutableList;
+
+import java.util.Optional;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.facebook.presto.hive.PartitionUpdate.FileWriteInfo;
+import static io.airlift.slice.Slices.utf8Slice;
+
+public class HiveManifestUtils
+{
+    private HiveManifestUtils()
+    {
+    }
+
+    public static Page createFileStatisticsPage(long fileSize, long rowCount)
+    {
+        // FileStatistics page layout:
+        //
+        // fileSize   rowCount
+        //  X             X
+        PageBuilder statsPageBuilder = new PageBuilder(ImmutableList.of(BIGINT, BIGINT));
+        statsPageBuilder.declarePosition();
+        BIGINT.writeLong(statsPageBuilder.getBlockBuilder(0), fileSize);
+        BIGINT.writeLong(statsPageBuilder.getBlockBuilder(1), rowCount);
+
+        return statsPageBuilder.build();
+    }
+
+    public static Optional<Long> getFileSize(Optional<Page> statisticsPage)
```

Comment:
The input doesn't have to be optional. If the user use optional, then it is expected to have empty return.
Use `getFileSize(Page statisticsPage, int position)`. Check our `Page` interface. `position` is very important as users can pass a large page (with many rows) and just get the one wanted.

Reply:
The statistics page should only have 1 row , because it holds the statistics of only one file. We should probably throw exception when there are more than 1 row right ? why should we have the `position` parameter ?

I mean we should throw exception if users send pages with more than 1 row. WDYT ?

Reply:
This is for future genericity. When we design interfaces, we need to make sure the interface is general